packages/back-end/src/routers/demo-datasource-project/demo-datasource-project.controller.ts

```diff
@@ -0,0 +1,336 @@
+    type: "binomial",
+    conversionDelayHours: 168,
+    conversionWindowHours: 168,
+    sql:
+      "SELECT\nuserId AS user_id,\ntimestamp AS timestamp\nFROM pages WHERE path = '/'",
+  },
+  {
+    name: "Days Active in Next 7 Days",
+    description:
+      "Count of times the user was active in the next 7 days after exposure",
+    type: "count",
+    conversionWindowHours: 168,
+    aggregation: "COUNT(DISTINCT value)",
+    sql:
+      "SELECT\nuserId AS user_id,\ntimestamp AS timestamp,\nDATE_TRUNC('day', timestamp) AS value\nFROM pages WHERE path = '/'",
+  },
+];
+
+const DEMO_RATIO_METRIC: Pick<
+  MetricInterface,
+  "name" | "description" | "type" | "sql"
+> = {
+  name: "Purchases - Average Order Value (ratio)",
+  description:
+    "The average value of purchases made in the 72 hours after exposure divided by the total number of purchases",
+  type: "revenue",
+  sql:
+    "SELECT\nuserId AS user_id,\ntimestamp AS timestamp,\namount AS value\nFROM orders",
+};
+
+// Experiment constants
+const DEMO_EXPERIMENTS: Pick<
+  ExperimentInterface,
+  "name" | "trackingKey" | "variations"
+>[] = [
+  {
+    name: "checkout-layout",
+    trackingKey: "checkout-layout",
+    variations: [
+      {
+        id: "0",
+        key: "0",
+        name: "Control",
+        screenshots: [],
+      },
+      {
+        id: "0",
+        key: "1",
+        name: "Compact",
+        screenshots: [],
+      },
+      {
+        id: "0",
+        key: "2",
+        name: "Spaced Out",
+        screenshots: [],
+      },
+    ],
+  },
+  {
+    name: "price-display",
+    trackingKey: "price-display",
+    variations: [
+      {
+        id: "0",
+        key: "0",
+        name: "Control",
+        screenshots: [],
+      },
+      {
+        id: "0",
+        key: "1",
+        name: "Hide discount",
+        screenshots: [],
+      },
+    ],
+  },
+];
+
+/**
+ * END Constants for Demo Datasource
+ */
+
+// region POST /demo-datasource-project
+
+type CreateDemoDatasourceProjectRequest = AuthRequest;
+
+type CreateDemoDatasourceProjectResponse = {
+  status: 200;
+  project: ProjectInterface;
+};
+
+/**
+ * POST /demo-datasource-project
+ * Create a demo-datasource-project resource
+ * @param req
+ * @param res
+ */
+export const postDemoDatasourceProject = async (
+  req: CreateDemoDatasourceProjectRequest,
+  res: Response<
+    CreateDemoDatasourceProjectResponse | PrivateApiErrorResponse,
+    EventAuditUserForResponseLocals
+  >
+) => {
+  req.checkPermissions("manageProjects", "");
+  req.checkPermissions("createDatasources", "");
+  req.checkPermissions("createMetrics", "");
+  req.checkPermissions("createAnalyses", "");
+
+  const { org } = getOrgFromReq(req);
+
+  try {
+    // TODO smarter management for cases when project exists
+    // TODO throw appropriate error after each step
+    await deleteProjectById(
+      getDemoDatasourceProjectIdForOrganization(org.id),
+      org.id
+    );
```

Comment:
I don't think the post endpoint should be deleting the project. We can use the delete project endpoint for that.

Reply:
So should I catch an error if we try to create the project and it already exists? Or should we just rely on us never doing that.

Reply:
If our data modelling is correct, we should have a uniqueness constraint on the project ID. Have you tried to see what would happen if you try to create it? If it fails, we should have a global error handler it falls back on. I think that's the approach we're using for everything. Create endpoints should not perform delete operations.

Reply:
Yeah, it is handled by this error: Error: Failed to create demo datasource and project with message: E11000 duplicate key error collection: test.projects index: id_1 dup key: { id: "prj_org_2s2jlyqlbl7tcr3_demo-datasource-project" }

If the endpoint only gets exposed when the project doesn't exist, I don't mind just throwing this generic error.

Reply:
That error is probably fine. It has Mongo or Mongoose error codes in it which is not ideal but it's the way we've been doing it everywhere else so I'm fine with it.